schemas/cloud-extra-public-ip.sql

```diff
@@ -0,0 +1,102 @@
+-- This schema adds the tables necessary for IP accounting as a seperate
+-- resource as part of the wider Cloud Accounting system.
+
+-- IPRecords
+CREATE TABLE IPRecords (
+  UpdateTime TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  MeasurementTime DATETIME NOT NULL,
+  MeasurementMonth INT NOT NULL,
+  MeasurementYear INT NOT NULL,
+
+  SiteID INT NOT NULL,
+  CloudComputeServiceID INT,
+  CloudType VARCHAR(255) NOT NULL,
+
+  LocalUserId VARCHAR(255) NOT NULL,
+  LocalGroupId VARCHAR(255) NOT NULL,
+
+  GlobalUserNameID INT NOT NULL,
+  FQAN VARCHAR(255) NOT NULL,
+  VOID INT NOT NULL,
+  VOGroupID INT NOT NULL,
+  VORoleID INT NOT NULL,
+
+  IPVersion TINYINT NOT NULL,
+  IPCount INT NOT NULL,
+
+  PublisherDNID INT NOT NULL,
+
+  -- We want to keep one record per site, per cloud, per user, per IP version,
+  -- per month/year.

```

Comment:
As the UsageRecord 'should be a “snapshot” of the number of IPs currently assigned to a user' [[1](https://egi-federated-cloud.readthedocs.io/en/latest/federation.html?highlight=public%20ip%20accounting#public-ip-usage-record)], maybe we should keep one record per day and build a monthly summary out of those daily records. 

Otherwise, what would we do if a User had 1 IP on Jan 1st, and 100 IPs on Jan 31st, and 0 in between?

Granted that case can only really be solved by using the timestamp in the key... otherwise you can just say "what would we do if a User had 1 IP at 9am on Jan 1st, and 100 IPs at 11pm on Jan 1st, and 0 in between?". Maybe one for the FedCloud to weigh in on.

Reply:
Reporting a combination of minimum, maximum and average number of IPs assigned to a user in a given accounting period (month) seems like the way forward, which probably means keeping 1 record per day in order to calculate those values.